packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -259,1548 +169,327 @@ class UpdatePackagesCommand extends FlutterCommand {
-    PubspecDependency? lastDependency;
-    for (int index = 0; index < lines.length; index += 1) {
-      String line = lines[index];
-      if (lastDependency == null) {
-        // First we look to see if we're transitioning to a new top-level section.
-        // The PubspecHeader.parse static method can recognize those headers.
-        final PubspecHeader? header = PubspecHeader.parse(line); // See if it's a header.
-        if (header != null) {
-          // It is!
-          section = header.section; // The parser determined what kind of section it is.
-          if (section == Section.header) {
-            if (header.name == 'name') {
-              packageName = header.value;
-            } else if (header.name == 'version') {
-              packageVersion = header.value;
-            }
-          } else if (section == Section.dependencies) {
-            // If we're entering the "dependencies" section, we want to make sure that
-            // it's the first section (of those we care about) that we've seen so far.
-            if (seenMain) {
-              throwToolExit(
-                'Two dependencies sections found in $filename. There should only be one.',
-              );
-            }
-            if (seenDev) {
-              throwToolExit(
-                'The dependencies section was after the dev_dependencies section in $filename. '
-                'To enable one-pass processing, the dependencies section must come before the '
-                'dev_dependencies section.',
-              );
-            }
-            seenMain = true;
-          } else if (section == Section.devDependencies) {
-            // Similarly, if we're entering the dev_dependencies section, we should verify
-            // that we've not seen one already.
-            if (seenDev) {
-              throwToolExit(
-                'Two dev_dependencies sections found in $filename. There should only be one.',
-              );
-            }
-            seenDev = true;
-          }
-          result.add(header);
-        } else if (section == Section.builders) {
-          // Do nothing.
-          // This line isn't a section header, and we're not in a section we care about.
-          // We just stick the line into the output unmodified.
-          result.add(PubspecLine(line));
-        } else if (section == Section.other) {
-          if (line.contains(kDependencyChecksum)) {
-            // This is the pubspec checksum. After computing it, we remove it from the output data
-            // since it will be recomputed later.
-            checksum = PubspecChecksum.parse(line);
+    final Map<String, Map<String, String>> deps = _mergeDeps(oldDeps, newDeps);
+    tempDir.deleteSync(recursive: true);
+    return deps;
+  }
+
+  void _relaxDeps(YamlEditor yamlEditor, RelaxMode relaxMode, Map<String, String> fixedDeps) {
+    for (final String depType in <String>['dependencies', 'dev_dependencies']) {
+      final Map<dynamic, dynamic> map =
```

Comment:
Can we use `Map<Object?, Object?>` instead? we should avoid using `dynamic` when possible.